src/components/map/Map.tsx

```diff
@@ -217,7 +224,65 @@ export const Map = ({
 };
 /* eslint-enable complexity */
 
+type TCluster = {
+  clusterColor: string;
+  geometry: { coordinates: [number, number] };
+  id: string;
+  onPress: () => void;
+  properties: { point_count: number };
+};
+
+const renderCluster = (cluster: TCluster) => {
+  const { clusterColor: backgroundColor, geometry, id, onPress, properties = {} } = cluster;
+  const { point_count: points } = properties;
+
+  const circleSizes = [60, 50, 40, 30];
+
+  return (
+    <Marker
+      key={`cluster-${id}`}
+      coordinate={{
+        longitude: geometry.coordinates[0],
+        latitude: geometry.coordinates[1]
+      }}
+      style={styles.clusterMarker}
+      onPress={onPress}
+    >
+      {circleSizes.map((size, index) => (
+        <View
+          key={`circle-${index}`}
+          style={[
+            styles.clusterCircle,
+            {
+              backgroundColor,
+              borderRadius: normalize(size / 2),
+              height: normalize(size),
+              opacity: 0.2 * (index + 1),
+              width: normalize(size)
+            }
+          ]}
+        >
+          <RegularText lightest center smallest>
+            {points}
+          </RegularText>
+        </View>
+      ))}
+    </Marker>
+  );
+};
```

Comment:
it would have been more following best practices to have the type and component before the main component in this file or having an own file. it blows this file up, right?!

Reply:
Yes. I didn't think of that. I move it between the main component and `MapIcon`. 👍

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/3a3849e76a3817a01a2a8dd4fd2193aa0f7e0b5c